beacon_node/network/src/sync/range_sync/batch.rs

```diff
@@ -75,9 +76,12 @@ impl<T: EthSpec> Batch<T> {
     pub fn to_blocks_by_range_request(&self) -> BlocksByRangeRequest {
         BlocksByRangeRequest {
             start_slot: self.start_slot.into(),
-            count: std::cmp::min(
-                T::slots_per_epoch() * EPOCHS_PER_BATCH,
-                self.end_slot.sub(self.start_slot).into(),
+            count: min(
+                min(
+                    T::slots_per_epoch() * EPOCHS_PER_BATCH,
+                    self.end_slot.sub(self.start_slot).into(),
+                ),
+                MAX_REQUEST_BLOCKS,
```

Comment:
If `T::slots_per_epoch() * EPOCHS_PER_BATCH > MAX_REQUEST_BLOCKS` this will cause syncing to break I'm pretty sure. 
We should add an assert to ensure this is never the case. Perhaps a `lazy_static!` is useful here for checking this.
We can then safely remove this extra min() check.